datatypes/src/operations/image/colorizer.rs

```diff
@@ -728,7 +833,49 @@ mod tests {
                     "2": [0, 0, 0, 255]
                 },
                 "noDataColor": [0, 0, 0, 0],
-                "defaultColor": [0, 0, 0, 0]
+                "defaultColor": [0, 0, 0, 0],
+                "overColor": null,
+                "underColor": null
+            })
+        );
+
+        assert_eq!(
+            serde_json::from_str::<Colorizer>(&serialized_colorizer.to_string()).unwrap(),
+            colorizer
+        );
+    }
+
+    #[test]
+    fn serialized_palette_with_extrema() {
+        let colorizer = Colorizer::palette(
+            [
+                (1.0.try_into().unwrap(), RgbaColor::white()),
+                (2.0.try_into().unwrap(), RgbaColor::black()),
+            ]
+            .iter()
+            .copied()
+            .collect(),
+            RgbaColor::transparent(),
+            RgbaColor::transparent(),
+            Some(RgbaColor::pink()),
+            Some(RgbaColor::pink()),
+        )
+        .unwrap();
+
+        let serialized_colorizer = serde_json::to_value(&colorizer).unwrap();
+
+        assert_eq!(
+            serialized_colorizer,
+            serde_json::json!({
+                "type": "palette",
+                "colors": {
+                    "1": [255, 255, 255, 255],
+                    "2": [0, 0, 0, 255]
+                },
+                "noDataColor": [0, 0, 0, 0],
+                "defaultColor": [0, 0, 0, 0],
+                "overColor": [255, 0, 255, 255],
+                "underColor": [255, 0, 255, 255]
```

Comment:
Somehow, only one should be set. This should not contradict.

Reply:
This is now being tested with two distinct operations.